### ISSUE TITLE
Data not being stored using KeyvRedis - Issue Typescript 

### ISSUE THREAD

**Issue by kishan-getstarted** [jaredwray/keyv#892]
I am currently using the Keyv library with the KeyvRedis store to manage data storage. However, I have encountered an issue where the data is not being stored as expected.

Code Snippet

`import KeyvRedis from '@keyv/redis';
import Keyv from 'keyv';

const keyvRedis = new KeyvRedis('redis://:@localhost:6379');
const keyv = new Keyv({ store: keyvRedis });
keyv.on('error', (err: any) => console.log('Connection Error', err));
async function main() {
  await keyvRedis.set('foo', 'baar');
}

main();`

Expected Behavior
I expect the code provided above to store the data ('foo', 'baar') using the KeyvRedis store.

Actual Behavior
Unfortunately, the data is not being stored as expected, and I am unable to retrieve it using the Keyv library.

Steps to Reproduce
Install the Keyv library and KeyvRedis store.
Set up a Redis server at localhost:6379.
Run the provided code snippet.
Additional Information
Keyv Version: ^4.5.3
Keyv Redis Version: ^2.7.0
Node.js Version: v18.12.0
Redis Version:  Redis 7.0.12 (00000000/0) 64 bit
Possible Solution
I have checked the documentation and ensured that my code follows the recommended usage. I suspect there might be an issue with the KeyvRedis integrationwith typescript. Any guidance on resolving this issue would be greatly appreciated.

<img width="1378" alt="Screenshot 2023-08-07 at 12 41 16 PM" src="https://github.com/jaredwray/keyv/assets/74973980/f2d45f93-ea89-4731-8f74-a02b3de60a9e">

**Comment by jaredwray** [comment]
Thanks @kishan-getstarted and I am looking into this as so far all the unit tests are setting and retrieving the data correctly.

**Comment by jaredwray** [comment]
if you remove the `.on` does it work?

**Comment by kishan-getstarted** [comment]
Nope,  but with the use of my good friend (GPT lol) I found the workaround...

`const Keyv = require('keyv');

// One of the following
const keyv = new Keyv('redis://:@localhost:6379');
console.log("🚀 ~ file: index.js:5 ~ keyv:", keyv)
keyv.on('error', err => console.log('Connection Error', err));

async function main() {
await keyv.set('test', 'never expires'); // true
await keyv.get('test'); // 'never expires'
}

main()`


@jaredwray still didn't understand why norma input was not working..
